File: src/models/tools/wikipedia.py
from typing import ClassVar, Dict, Literal
from pydantic import BaseModel


class WikipediaTool(BaseModel):
    config_id: ClassVar[str] = "wikipedia.WikipediaToolSpec"
    name: Literal["wikipedia"] = "wikipedia"
    tool_type: Literal["llamahub"] = "llamahub"
    label: Literal["Wikipedia"] = "Wikipedia"
    description: str = (
        "Use Wikipedia to gather more information about a topic from the query."
    )
    config: Dict = {}
    enabled: bool = False

File: src/ragapp/admin/client/tools/wikipedia.ts
import { z } from "zod";

export const WikipediaToolConfig = z.object({
  name: z.literal("wikipedia"),
  label: z.string().nullable().optional(),
  description: z.string().nullable().optional(),
  enabled: z.boolean().nullable().optional(),
  config: z.object({}).nullable().optional(),
});
export type WikipediaToolConfigType = z.infer<typeof WikipediaToolConfig>;
export const DEFAULT_WIKIPEDIA_TOOL_CONFIG = {
  label: "Wikipedia",
  description: "",
  config: {},
  enabled: false,
};
